code/game/machinery/computer/communications.dm

```diff
@@ -70,85 +72,76 @@
 				feedback_inc("alert_comms_blue",1)
 	tmp_alertlevel = 0
 
-/obj/machinery/computer/communications/Topic(href, href_list)
-	if(..(href, href_list))
-		return 1
-
-	if(!is_secure_level(src.z))
+/obj/machinery/computer/communications/tgui_act(action, params)
+	if(..())
+		return
+	if(!is_secure_level(z))
 		to_chat(usr, "<span class='warning'>Unable to establish a connection: You're too far away from the station!</span>")
-		return 1
+		return
+
+	. = TRUE
 
-	if(href_list["login"])
+	if(action == "auth")
 		if(!ishuman(usr))
 			to_chat(usr, "<span class='warning'>Access denied.</span>")
+			return FALSE
+		// Logout function.
+		if(authenticated != COMM_AUTHENTICATION_NONE)
+			authenticated = COMM_AUTHENTICATION_NONE
+			crew_announcement.announcer = null
+			setMenuState(usr, COMM_SCREEN_MAIN)
 			return
-
+		// Login function.
 		var/list/access = usr.get_access()
 		if(allowed(usr))
 			authenticated = COMM_AUTHENTICATION_MIN
-
 		if(ACCESS_CAPTAIN in access)
 			authenticated = COMM_AUTHENTICATION_MAX
 			var/mob/living/carbon/human/H = usr
 			var/obj/item/card/id = H.get_idcard(TRUE)
 			if(istype(id))
 				crew_announcement.announcer = GetNameAndAssignmentFromId(id)
-
-		SSnanoui.update_uis(src)
-		return
-
-	if(href_list["logout"])
-		authenticated = COMM_AUTHENTICATION_NONE
-		crew_announcement.announcer = ""
-		setMenuState(usr,COMM_SCREEN_MAIN)
-		SSnanoui.update_uis(src)
+		if(authenticated == COMM_AUTHENTICATION_NONE)
+			to_chat(usr, "<span class='warning'>You need to swipe your ID.</span>")
```

Comment:
I feel `swipe` here is misleading, given how this console doesnt actually handle being hit by an ID